src/ATen/native/xpu/SpectralOps.cpp

```diff
@@ -79,4 +79,39 @@ Tensor& _fft_c2r_xpu_out(
 #endif // USE_ONEMKL
 }
 
+Tensor _fft_r2c_xpu(
+    const Tensor& self,
+    IntArrayRef dim,
+    int64_t normalization,
+    bool onesided) {
+  TORCH_CHECK(self.is_floating_point());
+
+#if defined(USE_ONEMKL)
```

Comment:
`USE_ONEMKL` seems to be a misleading name. Basically you wrote - if `USE_ONEMKL` is defined then calll onemkl function (`xpu::_fft_r2c_mkl`), if it's not defined - still call onemkl function (`_fft_r2c_mkl`). This does not make any sense to me. Maybe you meant something like `USE_ONEMKL_XPU`, i.e. if onemkl was built with XPU support?

Reply:
@dvrogozh Thanks for the reminder! Your understanding is right. The motivation of `USE_ONEMKL` is to control oneMKL XPU support. The initial idea was to differentiate this switch from `USE_MKL` in stock Pytorch.
Will refine it.